NetKet/Machine/machine.hpp

```diff
@@ -77,6 +78,8 @@ class Machine : public AbstractMachine<T> {
       m_ = Ptype(new RbmSpin<T>(hilbert, pars));
     } else if (pars["Machine"]["Name"] == "RbmMultival") {
       m_ = Ptype(new RbmMultival<T>(hilbert, pars));
+    } else if (pars["Machine"]["Name"] == "FFNN") {
+      m_ = Ptype(new FFNN<T>(hilbert, pars));
```

Comment:
Also minor: If "Machine.Name" is neither of those values then `m_` will not be set which is an error. This should not happen in practise (as `CheckInput` should be called before passing the input to this function) but I'd add an `assert(CheckInput(pars));` at the start of this function to make sure this never happens (in release mode, the assert will not be compiled so there is no performance problem from checking the input multiple times there).